### ISSUE TITLE
Training cellpose model and Making masks

### ISSUE THREAD

**Issue by shan-stat** [MouseLand/cellpose#415]
I am struggling with training cellpose model by "models.CellposeModel.train(train_data, train_labels)" function. 
For the train_data, my image is RGB format and consists of (1040, 1392, 3), and for train_labels, the resulting mask from _seg.npy consists of (1040, 1392). An example is as follow: 

<img width="925" alt="Screen Shot 2021-12-24 at 10 02 34 AM" src="https://user-images.githubusercontent.com/64387506/147368074-eeac1df3-28d4-47d3-9b3a-7c2423512c5e.png">

When I tried "models.CellposeModel.train(train_data, train_labels)", I got the following error. I did not quite get it because the image and the mask are all at least of 2-D?

<img width="1332" alt="Screen Shot 2021-12-24 at 10 05 22 AM" src="https://user-images.githubusercontent.com/64387506/147368189-c3f242a1-590e-46b3-a3ec-f77807945a0b.png">

I am not sure if the way of making masks is correct. Could anyone help me?

**Comment by songzeji** [comment]
I am also having problems with training. 

When I train with the cyto2 pre-trained model I get the following error message:

```
/usr/local/lib/python3.7/dist-packages/torch/nn/modules/loss.py:520: UserWarning: Using a target size (torch.Size([8, 518, 224, 224])) that is different to the input size (torch.Size([8, 2, 224, 224])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
  return F.mse_loss(input, target, reduction=self.reduction)
Traceback (most recent call last):
  File "/usr/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.7/dist-packages/cellpose/__main__.py", line 436, in <module>
    main()
  File "/usr/local/lib/python3.7/dist-packages/cellpose/__main__.py", line 415, in main
    batch_size=args.batch_size, omni=args.omni)
  File "/usr/local/lib/python3.7/dist-packages/cellpose/models.py", line 850, in train
    batch_size, rescale)
  File "/usr/local/lib/python3.7/dist-packages/cellpose/core.py", line 934, in _train_net
    train_loss = self._train_step(imgi, lbl)
  File "/usr/local/lib/python3.7/dist-packages/cellpose/core.py", line 770, in _train_step
    loss = self.loss_fn(lbl,y)
  File "/usr/local/lib/python3.7/dist-packages/cellpose/models.py", line 759, in loss_fn
    loss = self.criterion(y[:,:2] , veci) 
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/loss.py", line 520, in forward
    return F.mse_loss(input, target, reduction=self.reduction)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/functional.py", line 3111, in mse_loss
    expanded_input, expanded_target = torch.broadcast_tensors(input, target)
  File "/usr/local/lib/python3.7/dist-packages/torch/functional.py", line 72, in broadcast_tensors
    return _VF.broadcast_tensors(tensors)  # type: ignore[attr-defined]
RuntimeError: The size of tensor a (2) must match the size of tensor b (518) at non-singleton dimension 1
```

Both my input training images and masks are 8-bit grayscale PNGs with shape [704, 520], and the batch size is 8.

Please could someone kindly tell me what the error means and how could I resolve it? I would be really grateful for your help!!

**Comment by kevinjohncutler** [comment]
@songzeji this may be a python version issue, as I see you are using 3.7. Have you tried setting up a conda environment with 3.8.5? 

@shan-stat I'm not sure what could be the issue, but try training using the command line tools instead of python. That will give us a lot more info. I recall having issues training cellpose via calling the function via python.

**Comment by carsen-stringer** [comment]
I agree with @kevinjohncutler's comments. please try training with the command line, which makes sure that the channels are the first dimension and takes care of making the flows. in the future we can add a notebook for training. if the CLI doesn't work please comment here and I'll reopen the issue.

**Comment by minerharry** [comment]
I'm having a similar issue as @songzeji, using the new notebook. I'm using images of size 672 x 512.
`creating new log file
2022-02-08 20:35:48,112 [INFO] WRITING LOG OUTPUT TO /root/.cellpose/run.log
2022-02-08 20:35:48,169 [INFO] TORCH CUDA version not installed/working.
2022-02-08 20:35:48,169 [INFO] >>>> using CPU
2022-02-08 20:35:48,169 [INFO] Downloading: "https://www.cellpose.org/models/cyto2torch_0" to /root/.cellpose/models/cyto2torch_0

100% 25.3M/25.3M [00:00<00:00, 58.1MB/s]
Not all flows are present. Run flow generation again.
Not all flows are present. Run flow generation again.
2022-02-08 20:35:48,916 [INFO] >>>> pretrained model /root/.cellpose/models/cyto2torch_0 is being used
2022-02-08 20:35:48,916 [INFO] >>>> during training rescaling images to fixed diameter of 30.0 pixels
2022-02-08 20:35:49,195 [INFO] Training with rescale = 1.00
2022-02-08 20:35:49,901 [INFO] flows precomputed
2022-02-08 20:35:49,988 [INFO] flows precomputed
2022-02-08 20:35:49,998 [WARNING] 29 train images with number of masks less than min_train_masks (0), removing from train set
/usr/local/lib/python3.7/dist-packages/numpy/core/fromnumeric.py:3441: RuntimeWarning: Mean of empty slice.
  out=out, **kwargs)
/usr/local/lib/python3.7/dist-packages/numpy/core/_methods.py:189: RuntimeWarning: invalid value encountered in double_scalars
  ret = ret.dtype.type(ret / rcount)
2022-02-08 20:35:50,009 [INFO] >>>> median diameter set to = 30
2022-02-08 20:35:50,009 [INFO] >>>> training network with 2 channel input <<<<
2022-02-08 20:35:50,009 [INFO] >>>> LR: 0.00020, batch_size: 8, weight_decay: 0.00001
2022-02-08 20:35:50,009 [INFO] >>>> ntrain = 9, ntest = 4
2022-02-08 20:35:50,009 [INFO] >>>> nimg_per_epoch = 9
/usr/local/lib/python3.7/dist-packages/torch/nn/modules/loss.py:520: UserWarning: Using a target size (torch.Size([8, 510, 224, 224])) that is different to the input size (torch.Size([8, 2, 224, 224])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
  return F.mse_loss(input, target, reduction=self.reduction)
Traceback (most recent call last):
  File "/usr/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/usr/local/lib/python3.7/dist-packages/cellpose/__main__.py", line 458, in <module>
    main()
  File "/usr/local/lib/python3.7/dist-packages/cellpose/__main__.py", line 437, in main
    omni=args.omni)
  File "/usr/local/lib/python3.7/dist-packages/cellpose/models.py", line 884, in train
    rescale=rescale, netstr=netstr)
  File "/usr/local/lib/python3.7/dist-packages/cellpose/core.py", line 946, in _train_net
    train_loss = self._train_step(imgi, lbl)
  File "/usr/local/lib/python3.7/dist-packages/cellpose/core.py", line 770, in _train_step
    loss = self.loss_fn(lbl,y)
  File "/usr/local/lib/python3.7/dist-packages/cellpose/models.py", line 771, in loss_fn
    loss = self.criterion(y[:,:2] , veci) 
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/modules/loss.py", line 520, in forward
    return F.mse_loss(input, target, reduction=self.reduction)
  File "/usr/local/lib/python3.7/dist-packages/torch/nn/functional.py", line 3111, in mse_loss
    expanded_input, expanded_target = torch.broadcast_tensors(input, target)
  File "/usr/local/lib/python3.7/dist-packages/torch/functional.py", line 72, in broadcast_tensors
    return _VF.broadcast_tensors(tensors)  # type: ignore[attr-defined]
RuntimeError: The size of tensor a (2) must match the size of tensor b (510) at non-singleton dimension 1`

**Comment by ChanLoongSheh** [comment]
I've got the same problem, I transform the bit and color space of the mask(int8 to int16, RGB to GRAY), then it works, but in the continuing training process, the calculated flow diagrams are wrong. #446
